Cork/App State.swift

```diff
@@ -119,4 +104,21 @@ class AppState: ObservableObject {
         
         sendNotification(title: String(localized: "notification.upgrade-process-started"))
     }
+    
+    func setCorruptedPackage(_ name: String) {
+        corruptedPackage = name
+        fatalAlertType = .installedPackageHasNoVersions
+        isShowingFatalError = true 
+    }
+    
+    func setCouldNotParseTopPackages() {
+        fatalAlertType = .couldNotParseTopPackages
+        isShowingFatalError = true
+    }
+}
+
+private extension UNUserNotificationCenter {
+    func authorizationStatus() async -> UNAuthorizationStatus {
+        await notificationSettings().authorizationStatus
+    }
```

Comment:
Moving this to an extension allows us to return the `authorizationStatus` without passing the UNNotificationSettings across a concurrency boundary.